DNN Platform/DotNetNuke.Web.Client/MvcClientResourceManager.cs

```diff
@@ -0,0 +1,588 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information
+
+// ReSharper disable once CheckNamespace
+namespace DotNetNuke.Web.Client.ClientResourceManagement
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Threading;
+    using System.Web;
+    using System.Web.Hosting;
+    using System.Web.Mvc;
+    using System.Xml;
+
+    using ClientDependency.Core;
+    using ClientDependency.Core.CompositeFiles.Providers;
+    using ClientDependency.Core.Config;
+    using ClientDependency.Core.Mvc;
+    using DotNetNuke.Instrumentation;
+    using DotNetNuke.Internal.SourceGenerators;
+
+    /// <summary>Provides the ability to request that client resources (JavaScript and CSS) be loaded on the client browser.</summary>
+    public partial class MvcClientResourceManager
+    {
+        /// <summary>The default css provider.</summary>
+        internal const string DefaultCssProvider = "DnnPageHeaderProvider";
+
+        /// <summary>The default javascript provider.</summary>
+        internal const string DefaultJsProvider = "DnnBodyProvider";
+
+        private static readonly ILog Logger = LoggerSource.Instance.GetLogger(typeof(ClientResourceManager));
+        private static readonly Dictionary<string, bool> FileExistsCache = new Dictionary<string, bool>();
+        private static readonly ReaderWriterLockSlim LockFileExistsCache = new ReaderWriterLockSlim();
+
+        /// <summary>Adds the necessary configuration to website root <c>web.config</c> to use the Client Dependency component.</summary>
+        public static void AddConfiguration()
```

Comment:
Why is this beind done as a Web.Client item, shouldn't this be handled by the existing functionality?  I'm concerned this is going to introduce a duplicate or race condition